scripts/review/diff.ts

```diff
@@ -0,0 +1,113 @@
+import util from 'node:util';
+import fs from 'node:fs/promises';
+import parseDiff from 'parse-diff';
+import assert from 'node:assert';
+
+type Options = {
+  diffPatchFileName: string;
+  verbose: boolean;
+};
+
+type Deployment = { [key: number]: string };
+
+function parseOptions(): Options {
+  const options = {
+    diffPatchFileName: { type: 'string' },
+    verbose: { type: 'boolean' },
+  } as const;
+  const { values } = util.parseArgs({ options });
+
+  const option = 'diffPatchFileName';
+  if (values[option] === undefined) {
+    throw new Error(`missing --${option} flag`);
+  }
+
+  return {
+    ...(values as Options),
+    verbose: values.verbose === true,
+  };
+}
+
+async function main() {
+  const options = parseOptions();
+  const debug = (...msg: unknown[]) => {
+    if (options.verbose) {
+      console.debug(...msg);
+    }
+  };
+
+  // Read the content of the files
+  const diffPatchContent = await fs.readFile(options.diffPatchFileName, 'utf-8');
+  const diffPatch = parseDiff(diffPatchContent);
+
+  let highestChainID = false;
+  let additionalDeploymentToSameChainID = false;
+
+  for (const { to, from, additions, deletions } of diffPatch) {
+    // Check to see if the file name was changed.
+    if (to !== from) {
+      throw new Error(`ERROR: file renamed to ${to} from ${from}`);
+    }
+
+    debug(`Checking ${to} with ${additions} additions and ${deletions} deletions`);
+
+    // Check to see if the changes are valid and set any edge case flags.
+    if (additions === 2 && deletions === 1) {
+      highestChainID = true;
+    } else if (additions === 1 && deletions === 1) {
+      additionalDeploymentToSameChainID = true;
+    } else if (additions !== 1 || deletions !== 0) {
+      throw new Error(`ERROR: ${to} has invalid changes`);
+    }
+  }
+
+  if (highestChainID) {
+    debug('Highest chain ID deployment');
+    for (const { chunks } of diffPatch) {
+      // Filter chunks as `changes` to only include items that are additions (add) or deletions (del).
+      const changes = chunks.flatMap(({ changes }) => changes.filter(({ type }) => type === 'add' || type === 'del'));
+      assert(changes.length === 3);
+      assert(changes.map((c) => c.type).join() === 'del,add,add');
```

Comment:
I'm not a fan of this, since it claims that `["del,add", "add"] (equals) ["del", "add,add"]`.

You can introduce a function for this:

```js
function hasChangeTypes(chunks, types) {
  return chunks.length === types.length
    && chunks.every(({ type }, i) => type === types[i]);
}
```

This can be reused below as well.

Reply:
Also, with this function, you don't need to check the length since it is implicit here.